Tests/test_content.py

```diff
@@ -1016,14 +1034,218 @@ def handle_github_response(response):
     return res_dict
 
 
+@contextmanager
+def acquire_test_lock(test_details: dict,
+                      default_test_timeout: int,
+                      prints_manager: ParallelPrintsManager,
+                      thread_index: int,
+                      is_not_mocked: bool = False) -> None:
+    """
+    This is a context manager that handles all the locking and unlocking of integrations.
+    Execution is as following:
+    * Attempts to lock the test's integrations and yields the result of this attempt
+    * If lock attempt has failed- puts the test back in the queue
+    * Once the test is done- will unlock all integrations
+    Args:
+        test_details: Details of the currently executed test
+        default_test_timeout: Default test timeout in seconds
+        prints_manager: ParallelPrintsManager object
+        thread_index: The index of the thread that executes the unlocking
+        is_not_mocked: If the test is mockable - no need to actually lock it
+    """
+    storage_client = storage.Client()
+    if is_not_mocked:
+        locked = lock_integrations(test_details, default_test_timeout, storage_client, prints_manager, thread_index)
+    else:
+        locked = True
+    try:
+        yield locked
+    finally:
+        if not locked or is_not_mocked:
+            return
+        # executing the test could take a while, re-instancing the storage client
+        storage_client = storage.Client()
+        unlock_integrations(get_integrations_list(test_details), prints_manager, storage_client, thread_index)
+
+
+def lock_integrations(test_details: dict,
+                      default_test_timeout: int,
+                      storage_client: storage.Client,
+                      prints_manager: ParallelPrintsManager,
+                      thread_index: int) -> bool:
+    """
+    Locks all the test's integrations
+    Args:
+        test_details: Details of the currently executed test
+        default_test_timeout: Default test timeout in seconds
+        storage_client: The GCP storage client
+        prints_manager: ParallelPrintsManager object
+        thread_index: The index of the thread that executes the unlocking
+
+    Returns:
+        True if all the test's integrations were successfully locked, else False
+    """
+    integrations = get_integrations_list(test_details)
+    if not integrations:
+        return True
+    existing_integrations_lock_files = get_locked_integrations(integrations, storage_client)
```

Comment:
existing makes it sound like all integrations in general
```suggestion
    current_integrations_lock_files = get_locked_integrations(integrations, storage_client)
```